### PR TITLE
Fix #68: Require auth for some pages using mixins

### LINKED ISSUE #68: Use a mixin instead of checking whether the user is authenticated in templates

### PR DESCRIPTION
**Pull request by openjck**
Raise a 403 when a user visits pages related to a push application they
didn't create. Move the push application landing page to /push and
require login to show it.

### CODE REVIEW

File: dashboard/views.py [hunk 1]
```diff
@@ -6,7 +7,19 @@
 from push.models import PushApplication
 
 
-class Home(TemplateView):
+class Login(TemplateView):
+    template_name = 'dashboard/login.html'
+
+
+class Status(TemplateView):
```
Comment (groovecoder) [inline dashboard/views.py]: Hmm ... I think we'd like to keep this on the home page? Or change this to a new `/push` url for a push home page.

File: dashboard/views.py [hunk 2]
```diff
@@ -6,7 +7,19 @@
...
+    template_name = 'dashboard/status.html'
+
+
+class PermissionDenied(TemplateView):
+    template_name = 'dashboard/errors/403.html'
+
+
+class Home(LoginRequiredMixin, TemplateView):
```
Comment (groovecoder) [inline dashboard/views.py]: We don't need the `LoginRequiredMixin` here do we? With it, as soon as I go to http://127.0.0.1:8000 I'm redirected to http://127.0.0.1:8000/accounts/login/?next=/

**Comment by groovecoder** [comment]
Coveralls wants you to write a test for the new `PushApplication.created_by()` method.

File: dashboard/views.py (hunk 1 above)
Comment (openjck) [inline dashboard/views.py]: Sure. I like the idea of having a separate _/push_ page for the form. That way, the page either loads completely or not at all depending on whether someone is logged in. We could even add an _Add push application_ button to the main navigation to take people there.

File: dashboard/views.py (hunk 2 above)
Comment (openjck) [inline dashboard/views.py]: Yeah, it only makes sense if everything on the homepage is behind login. If we change the homepage this won't make sense.

**Comment by groovecoder** [comment]
So, I'd say:
- Combine the `/status` page, the table of the user's push apps, and the form for creating a new push app all into a new `/push` page.
- Remove the `LoginRequiredMixin` from the home view, and simply show a login button and a link to the new `/push` page.
- Add a test for `PushApplication.created_by`

**Comment by openjck** [comment]
Updated

**Comment by groovecoder** [comment]
Looking good! I'll merge & rebase my stuff of this.

**Pull request merged**